playbook/runner.go

```diff
@@ -179,6 +205,15 @@ func HandleRun(ctx context.Context, run models.PlaybookRun) error {
 		return fmt.Errorf("failed to fetch playbook: %w", err)
 	}
 
+	filterPassed, err := checkPlaybookFilter(ctx, playbook, run)
```

Comment:
This check should be done in the parameter templating function so we don't event get to execution